rest/route/host_create.go

```diff
@@ -70,7 +70,40 @@ func (h *hostCreateHandler) Run(ctx context.Context) gimlet.Responder {
 	}
 
 	ids := []string{}
+	t, err := task.FindOneId(h.taskID)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "finding task '%s'", h.taskID))
+	}
+	if t == nil {
+		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("task '%s' not found", h.taskID),
+		})
+	}
+	initialHosts, err := host.FindHostsSpawnedByTask(ctx, h.taskID, t.Execution)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "finding hosts spawned by task '%s'", h.taskID))
+	}
+	numInitialHosts := len(initialHosts)
 	for i := 0; i < numHosts; i++ {
+		// If there are an unexpected number of host intents, log a message and continue.
+		spawnedHosts, err := host.FindHostsSpawnedByTask(ctx, h.taskID, t.Execution)
+		if err != nil {
+			return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "finding hosts spawned by task '%s'", h.taskID))
+		}
+
+		if len(spawnedHosts) != numInitialHosts+i {
+			grip.Debug(message.Fields{
```

Comment:
Debug logging is typically only for temporary debugging and people are allowed to remove debug logs later on. Do we plan to have this log stay around in the long term to diagnose problems, or is it just a temporary check?